src/python/WMCore/ACDC/DataCollectionService.py

```diff
@@ -350,3 +353,32 @@ def getLumilistWhitelist(self, collectionID, taskName):
 
         lumiList = LumiList(compactList=self.getLumiWhitelist(collectionID, taskName))
         return lumiList
+
+
+def getMergedParents(childLFNs):
+    import threading
+    from WMCore.DAOFactory import DAOFactory
+
+    myThread = threading.currentThread()
+    daoFactory = DAOFactory(package="WMCore.WMBS", logger=myThread.logger,
+                                dbinterface=myThread.dbi)
+
+    getParentInfoAction = daoFactory(classname="Files.GetParentInfo")
+
+    parentsInfo = getParentInfoAction.execute(childLFNs)
+    newParents = set()
+    unmergedParents = set()
+    for parentInfo in parentsInfo:
```

Comment:
Shouldn't we keep looking for parent and grand parents until one of them no longer exists (None)?
Eventually we might have taskChains with way too many tasks that could cause a lack of grandgrandgrand parentage information.

Reply:
It does. it will look for first merged parents. If it gets unmerged parents it will keep look until it finds merged parents or no parents.